editions/tw5.com/tiddlers/wikitext/Macro Definitions in WikiText.tid

```diff
@@ -17,6 +19,20 @@ Alternatively, the entire definition can be presented on a single line without a
 \define sayhi(name:"Bugs Bunny") Hi, I'm $name$.
 ```
 
+Macro definitions can be nested by specifying the name of the macro in the `\end` marker. For example:
+
+<<wikitext-example-without-html src:"""\define special-button(caption:"Click me")
+\define actions()
+<$action-sendmessage $message="tm-notify" $param="HelloThere"/>
+\end actions
+<$button actions=<<actions>>>
+$caption$
+</$button>
+\end special-button
+
+<<special-button>>
+""">>
+
```

Comment:
Question 1: Is this a requirement to have all nested macros right after `\define parent()`?
Of course, the current logic is fine and makes codes simpler to follow.

Question ii: Is it possible to allow one tab, or few whitespaces before `\define` and `\end` for nested macros?
I understood pragma `\define` shall start from first column!

Reply:
> Question 1: Is this a requirement to have all nested macros right after `\define parent()`? Of course, the current logic is fine and makes codes simpler to follow.

Yes – the contents of the parent macro forms a new parsing context that behaves just the same as when the entire body of a tiddler is parsed: pragmas still need to be at the top.

> Question ii: Is it possible to allow one tab, or few whitespaces before `\define` and `\end` for nested macros? I understood pragma `\define` shall start from first column!

I wondered about that, I guess it will be a popular request. I think we'd apply that to all pragmas if we were to change it.